Common/Tools/handleParamTPCResponse.cxx

```diff
@@ -0,0 +1,300 @@
+      }
+    }
+    if (endTime == 0) { // Default value for endTime
+      endTime = 4108971600000;
+    }
+  }
+
+  if (optMode.compare("read") == 0) { // Read existing object from local file
+    if (inFilename.empty()) {
+      LOG(error) << "read mode defined with no input file, please set --read-from-file";
+      return 1;
+    }
+
+    TFile fin(inFilename.data(), "READ");
+    if (!fin.IsOpen()) {
+      LOG(error) << "Input file " << inFilename << " could not be read";
+      return 1;
+    }
+
+    tpc.reset(fin.Get<Response>(objname.c_str()));
+    if (!tpc) {
+      LOG(error) << "Object with name " << objname << " could not be found in file " << inFilename;
+      return 1;
+    }
+    LOG(info) << "Reading existing TPCPIDResponse object " << objname << " from " << inFilename << ":";
+    tpc->PrintAll();
+    return 0;
+  }
+
+  else if (optMode.compare("write") == 0 || optMode.compare("push") == 0) // Create new object to write to local file or push to CCDB
+  {
+    if (!inFilename.empty()) { // Read from existing file to push to CCDB
+      LOG(info) << "Reading from existing file to write to CCDB:";
+      TFile fin(inFilename.data(), "READ");
+      if (!fin.IsOpen()) {
+        LOG(error) << "Input file " << inFilename << " could not be read";
+        return 1;
+      }
+      tpc.reset(fin.Get<Response>(inobjname.c_str()));
+      if (!tpc) {
+        LOG(error) << "Object with name " << objname << " could not be found in file " << inFilename;
+        return 1;
+      }
+      tpc->PrintAll();
+    } else { // Create new object if file not specified
+      LOG(info) << "Creating new TPCPIDResponse object with defined parameters:";
+
+      tpc.reset(new Response());
+      tpc->SetBetheBlochParams(BBparams);
+      tpc->SetResolutionParams(sigparamsGlobal);
+      tpc->SetMIP(mipval);
+      tpc->SetChargeFactor(chargefacval);
+      tpc->SetMultiplicityNormalization(multNormval);
+      tpc->SetUseDefaultResolutionParam(useDefaultParam);
+      tpc->PrintAll();
+    }
+    if (optMode.compare("write") == 0) {
+      if (outFilename.empty()) {
+        LOG(error) << "'write' mode specified, but no output filename. Quitting";
+        return 1;
+      }
+
+      LOG(info) << "Writing to output file " << outFilename;
+      TFile fout(outFilename.data(), "RECREATE");
+      if (!fout.IsOpen()) {
+        LOG(error) << "Output file " << inFilename << " could not be written to";
+        return 1;
+      }
+      fout.cd();
+      //   tpc->Print();
+      fout.WriteObject(tpc.get(), objname.c_str());
+      fout.Close();
+      LOG(info) << "File successfully written";
+      return 0;
+    }
+
+    if (optMode.compare("push") == 0) {
+      LOG(info) << "Attempting to push object to CCDB";
+
+      if (optDelete) {
+        api.truncate(pathCCDB);
+      }
+
+      if (runnumber != 0) {
+        metadata["runnumber"] = Form("%i", runnumber);
```

Comment:
Same comment here, this looks like taken from https://github.com/AliceO2Group/O2Physics/blob/master/Common/Tools/handleParamBase.h#L64 is there something wrong/missing that it had to be reimplemented?